src/transformers/modeling_utils.py

```diff
@@ -1927,6 +1936,23 @@ def from_pretrained(cls, pretrained_model_name_or_path: Optional[Union[str, os.P
             subfolder (`str`, *optional*, defaults to `""`):
                 In case the relevant files are located inside a subfolder of the model repo on huggingface.co, you can
                 specify the folder name here.
+            variant (`str`, *optional*):
+                If specified load weights with the `variant` suffix. If no weights with the `variant` suffix are
+                available an error is thrown.
+
+                <Tip>
+
+                Passing `variant` is useful to load weights serialized in float16. *E.g.*:
+
+                ```python
+                from transformers import AutoModelForCausalLM
+
+                model = AutoModelForCausalLM.from_pretrained("EleutherAI/gpt-j-6B", variant="float16")
+                ```
```

Comment:
This should use another repo. I don't think we will get the Eleutherai folks to upload a new version of the model in their repo since they never replied when I proposed to add a sharded version.